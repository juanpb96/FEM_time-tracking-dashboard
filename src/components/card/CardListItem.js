

export const CardListItem = ({selectedTimeframe, title, timeframes}) => {
    
    const { current, previous } = timeframes[selectedTimeframe];
    const formattedTimeframe = selectedTimeframe.split('ly')[0];

    return (
        <div className="card">
            <div className="card__content">
                <h3 className="card__title">{ title }</h3>
                <h2 className="card__current">{ current }hrs</h2>
                <p className="card__previous">Last { formattedTimeframe === 'dai' ? 'day' : formattedTimeframe } - { previous }hrs</p>
            </div>
        </div>
    );

};
