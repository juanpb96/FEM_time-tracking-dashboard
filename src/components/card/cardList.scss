@media screen and (min-width: 768px) {
    .card-list {
        display: flex;
        flex-wrap: wrap;
        gap: 30px;
    }
}