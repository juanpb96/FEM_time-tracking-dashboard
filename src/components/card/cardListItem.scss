@import '/public/variables.scss';

.card {
    position: relative;
    padding-top: 2.375rem;
    margin-top: 1.5rem;
    border-radius: $border-radius;
    overflow: hidden;

    @for $i from 1 through 6 {
        &:nth-child(#{$i}) {
            background: linear-gradient(nth($card-border-colors, $i), 80%, $dark-blue);
            
            &::before {
                content: nth($card-icons, $i);
                position: absolute;
                top: nth($card-top-icons, $i);
                right: 17px;
                z-index: 0;
            }
        }
    }

    &__content {
        position: relative;
        padding: 1.75rem 1.5rem;
        background-color: $dark-blue;
        border-radius: $border-radius;
        z-index: 1;
    }

    &__timeframe {
        display: flex;
        justify-content: space-between;
        align-items: center;
    }

    &__title,
    &__current {
        color: #FFF;
        margin: 0;
    }

    &__title {
        margin-bottom: 0.375rem;
        position: relative;
        font: $fw-medium 1.125rem/1.3313rem $ff-rubik;

        span {
            display: inline-block;
            max-width: 90%;
        }

        &::after {
            content: '';
            width: 21px;
            height: 5px;
            position: absolute;
            top: calc(50% - 2.5px);
            right: 0;
            background-image: url(~/public/assets/icon-ellipsis.svg);
            background-size: cover;
            cursor: pointer;
        }
    }

    &__current {
        font: $fw-light clamp(2rem, 3vw, 3.5rem)/2.375rem $ff-rubik;
    }

    &__previous {
        color: $pale-blue;
        font: $fw-regular 0.9375rem/1.1113rem $ff-rubik;
    }
}

@media only screen and (min-width: 768px) {
    .card {
        margin-top: 0;
        min-width: 255px;
        
        &__title {
            margin-bottom: 1.5rem;
        }

        &__timeframe {
            flex-direction: column;
            justify-content: flex-start;
            align-items: flex-start;
            gap: 0.5rem;
        }

        &__previous {
            margin: 0;
        }
    }
}

// TODO: Desktop styles