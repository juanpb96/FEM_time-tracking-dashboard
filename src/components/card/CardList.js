import { getData } from "../../selectors/getData";
import { CardListItem } from "./CardListItem";


export const CardList = ({timeframe}) => {

    const data = getData();

    return (
        <>
            <h2>CardList</h2>
            {
                data.map( activity => (
                    <CardListItem 
                        key={ activity.title }
                        selectedTimeframe={ timeframe }
                        {...activity}
                    />
                ))
            }
        </>
    );
};
