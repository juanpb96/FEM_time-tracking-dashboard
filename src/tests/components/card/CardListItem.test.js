import { shallow } from "enzyme";
import { CardListItem } from "../../../components/card/CardListItem";

const timeframes = {
    daily: {
        current: 5,
        previous: 7
    },
    weekly: {
        current: 32,
        previous: 36
    },
    monthly: {
        current: 103,
        previous: 128
    }
};

describe('test in <CardListItem />', () => { 

    test('should render the component correctly', () => { 

        const selectedTimeframe = 'weekly';
        const title = 'Work';

        const wrapper = shallow(
            <CardListItem 
                selectedTimeframe={ selectedTimeframe }
                title={ title }
                timeframes={ timeframes }
            />
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.card__title span').get(0).props.children ).toBe( title );

        const expectedPreviousMessage = `Last week - ${ timeframes[selectedTimeframe].previous }hrs`;

        expect( wrapper.find('.card__previous').text().trim() ).toBe( expectedPreviousMessage );

    });

    test('should display "Yesterday" if timeframe is "daily"', () => { 

        const selectedTimeframe = 'daily';
        const title = 'Work';

        const wrapper = shallow(
            <CardListItem 
                selectedTimeframe={ selectedTimeframe }
                title={ title }
                timeframes={ timeframes }
            />
        );

        const expectedPreviousMessage = `Yesterday - ${ timeframes[selectedTimeframe].previous }hrs`;

        expect( wrapper.find('.card__previous').text().trim() ).toBe( expectedPreviousMessage );

    })

});