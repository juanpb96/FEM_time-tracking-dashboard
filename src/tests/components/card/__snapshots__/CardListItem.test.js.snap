// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <CardListItem /> should render the component correctly 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <h3
      className="card__title"
    >
      <span>
        Work
      </span>
      <span
        className="card__icon-ellipsis"
      >
        <span
          className="sr-only"
        >
          See more
        </span>
      </span>
    </h3>
    <div
      className="card__timeframe"
    >
      <h2
        className="card__current"
      >
        32
        hrs
      </h2>
      <p
        className="card__previous"
      >
        Last week
         - 
        36
        hrs
      </p>
    </div>
  </div>
</div>
`;
